lib/content-hasher.js

```diff
@@ -0,0 +1,106 @@
+"use strict";
+
+
+/**
+ * @typedef {ContentHasher}
+ * @function push
+ * @function pushList
+ * @function hash
+ */
+
+
+/**
+ * @param {crypto} crypto
+ * @param {stream-to-promise} streamToPromise
+ * @param {stream-concat} StreamConcat
+ * @param {stream~Readable} StreamReadable
+ * @return {ContentHasher}
+ */
+module.exports = (crypto, streamToPromise, StreamConcat, StreamReadable) => {
+    /**
+     * Creates a Readable stream out of a string.
+     *
+     * @param {string} content
+     * @return {stream~Readable}
+     */
+    function createStream(content) {
+        var stream;
+
+        stream = new StreamReadable();
+        stream.push(content);
+        stream.push(null);
+
+        return stream;
+    }
+
+
+    /**
+     * Responsible for hashing a list of content. The real reason to
+     * use this class over a simple concated and hashed string is to
+     * allow streaming requests. The stream will be piped so you must
+     * push the content to another stream for the request.
+     */
+    class ContentHasher {
+        /**
+         * @param {string} secret
+         */
+        constructor(secret) {
+            this.contentList = [];
+            this.secret = secret;
+        }
+
+
+        /**
+         * Adds more content that will be used in the hash.
+         *
+         * @param {string|stream~Readable} content
```

Comment:
This has been updated to `@param {(string|Buffer|stream~Readable)} content`. Headers and everything gets pushed to this so it takes all three.